### PR TITLE
Added Swagger Spec for ManagedServices

### PR DESCRIPTION
**Pull request by ajay-sainy**
PR is already approved in private repo (https://github.com/Azure/azure-rest-api-specs-pr/pull/661)

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/managedservices/resource-manager/preview/2018-06-01-preview/managedservices.json [hunk 1]
```diff
@@ -0,0 +1,856 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Ok - Returns information about the updated registration definition.",
+            "schema": {
+              "$ref": "#/definitions/RegistrationDefinition"
+            }
+          },
+          "201": {
```
Comment (jianghaolu) [inline specification/managedservices/resource-manager/preview/2018-06-01-preview/managedservices.json]: Is this operation a long running operation? When would the API return 201?

Comment (ajay-sainy) [reply specification/managedservices/resource-manager/preview/2018-06-01-preview/managedservices.json]: No, this is not a long running opertation. It returns 201 when created and 200 when updated.

**Comment by ajay-sainy** [comment]
@jianghaolu Could you please let me know why the checks are failing and how can I fix?

File: specification/managedservices/resource-manager/readme.go.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  license-header: MICROSOFT_APACHE_NO_VERSION
+  namespace: managedservices
+  clear-output-folder: true
+```
+
+### Go multi-api
+
+``` yaml $(go) && $(multiapi)
+batch:
+  - tag: package-2018-06-01-preview
```
Comment (jianghaolu) [inline specification/managedservices/resource-manager/readme.go.md]: should be `package-2018-06-preview` per https://github.com/Azure/azure-rest-api-specs/pull/5328/files#diff-20f9c7f8021cc64c275ac54a421744dcR33

Comment (ajay-sainy) [reply specification/managedservices/resource-manager/readme.go.md]: Made the suggested change.

**Comment by ajay-sainy** [comment]
@jianghaolu Is the step **Automation for azure-sdk-for-net** supposed to fail for new RPs? If not, how can I fix this?

**Comment by shahabhijeet** [comment]
startbuild

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5438 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5438/commits/e4b89464326a9f5e9a646434ab346bf20a72e824

**Comment by ajay-sainy** [comment]
@jianghaolu Is anything else pending for this PR?

**Review by jianghaolu — APPROVED** [review]

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 3]
```diff
@@ -0,0 +1,845 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/{scope}/providers/Microsoft.ManagedServices/registrationDefinitions/{registratonDefinitionId}": {
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: registraton -> registration

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 4]
```diff
@@ -0,0 +1,845 @@
...
+          "RegistrationDefinitions"
+        ],
+        "operationId": "RegistrationDefinitions_Get",
+        "description": "Gets the registration definition details.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ScopeParameter"
+          },
+          {
+            "$ref": "#/parameters/RegistratonDefinitionIdParameter"
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: registraton -> registration

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 5]
```diff
@@ -0,0 +1,845 @@
...
+      },
+      "delete": {
+        "tags": [
+          "RegistrationDefinitions"
+        ],
+        "operationId": "RegistrationDefinitions_Delete",
+        "description": "Deletes the registration definition.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/RegistratonDefinitionIdParameter"
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: registraton -> registration

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 6]
```diff
@@ -0,0 +1,845 @@
...
+      },
+      "put": {
+        "tags": [
+          "RegistrationDefinitions"
+        ],
+        "operationId": "RegistrationDefinitions_CreateOrUpdate",
+        "description": "Creates or updates a registration definition.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/RegistratonDefinitionIdParameter"
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: registraton -> registration

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 7]
```diff
@@ -0,0 +1,845 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "Put Registration Definition": {
+            "$ref": "./examples/PutRegistrationDefinition.json"
+          }
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.ManagedServices/registrationAssignments/{registratonAssignmentId}": {
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: registraton -> registration

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 8]
```diff
@@ -0,0 +1,845 @@
...
+          "RegistrationAssignments"
+        ],
+        "operationId": "RegistrationAssignments_Get",
+        "description": "Gets the details of specified registration assignment.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ScopeParameter"
+          },
+          {
+            "$ref": "#/parameters/RegistratonAssignmentIdParameter"
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: registraton -> registration

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 9]
```diff
@@ -0,0 +1,845 @@
...
+          "RegistrationAssignments"
+        ],
+        "operationId": "RegistrationAssignments_Delete",
+        "description": "Deletes the specified registration assignment.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ScopeParameter"
+          },
+          {
+            "$ref": "#/parameters/RegistratonAssignmentIdParameter"
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: registraton -> registration

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 10]
```diff
@@ -0,0 +1,845 @@
...
+          "RegistrationAssignments"
+        ],
+        "operationId": "RegistrationAssignments_CreateOrUpdate",
+        "description": "Creates or updates a registration assignment.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ScopeParameter"
+          },
+          {
+            "$ref": "#/parameters/RegistratonAssignmentIdParameter"
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: registraton -> registration

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 11]
```diff
@@ -0,0 +1,845 @@
...
+    },
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "description": "The API version to use for this operation."
+    },
+    "RegistratonAssignmentIdParameter": {
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: registraton -> registration

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 12]
```diff
@@ -0,0 +1,845 @@
...
+      "name": "registratonAssignmentId",
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: registraton -> registration

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 13]
```diff
@@ -0,0 +1,845 @@
...
+      "in": "path",
+      "required": true,
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "description": "Guid of the registration assignment."
+    },
+    "RegistratonDefinitionIdParameter": {
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: registraton -> registration

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 14]
```diff
@@ -0,0 +1,845 @@
...
+      "name": "registratonDefinitionId",
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: registraton -> registration

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 15]
```diff
@@ -0,0 +1,845 @@
...
+          "description": {
+            "type": "string",
+            "description": "Description of the registration definition."
+          },
+          "authorizations": {
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/Authorization"
+            },
+            "description": "Authorization tuple containing prinicpal id of the user/security group or service principal and id of the build-in role."
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: prinicpal -> principal

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 16]
```diff
@@ -0,0 +1,845 @@
...
+                "description": {
+                  "type": "string",
+                  "description": "Description of the registration definition."
+                },
+                "authorizations": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/definitions/Authorization"
+                  },
+                  "description": "Authorization tuple containing prinicpal id of the user/security group or service principal and id of the build-in role."
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: prinicpal -> principal

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 17]
```diff
@@ -0,0 +1,845 @@
...
+            },
+            "description": {
+              "type": "string",
+              "description": "Description of the operation."
+            }
+          }
+        }
+      },
+      "type": "object",
+      "description": "Object that desctibes a single Microsoft.ManagedServices operation."
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: desctibes -> describes

File: specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json [hunk 18]
```diff
@@ -0,0 +1,845 @@
...
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "description": "Guid of the registration definition."
+    },
+    "ExpandRegistrationDefinition": {
+      "name": "$expandRegistrationDefinition",
+      "in": "query",
+      "x-ms-parameter-location": "method",
+      "type": "boolean",
+      "description": "Tells whether to return registration definition details also along with registration assingmnet details."
```
Comment (nschonni) [inline specification/managedservices/resource-manager/Microsoft.ManagedServices/preview/2018-06-01-preview/managedservices.json]: assingmnet -> assignment

**Pull request merged**